### ISSUE TITLE
Question about training on custom CNN

### ISSUE THREAD

**Issue by anvenkat09** [experiencor/basic-yolo-keras#149]
Hello, 

Wonderful implementation! I was following your step by step notebook and was wondering if I could just replace the yolo model you have with my own CNN with the same reshaped output and true boxes, and train from scratch using my own dataset? I'd like to try training without the use of pretrained weights just to see how the model converges.

**Comment by experiencor** [comment]
@anvenkat09 still got this problem?

**Comment by 11mhg** [comment]
Hey Experiencor,

I am actually getting a similar problem but on the recall. On certain images, it seems that the recall is simply 0. This affects the average recall and after a while, it simply becomes 0. I read somewhere that I should probably be looking into using clipnorm or clipvalue to clip the gradient from exploding and use a grid search to find proper values for that. 

What do you think? Am I doing something wrong?

Thanks :)

**Comment by anvenkat09** [comment]
@experiencor 
No I fixed it. Don't remember exactly how, but it's not an issue anymore. It had something to do with my pretrained weights

**Comment by experiencor** [comment]
@11mhg Recall approaching 0 is another problem. It normally happens during warm-up, but not actual training. Please refer to the readme for tips.

**Comment by 11mhg** [comment]
Hey @experiencor , I followed the readme you have, and I noticed I'm still receiving this problem. I allowed the net to perform warmup and then I did proper training and the actual recall never changed from 0.

**Comment by experiencor** [comment]
@11mhg What dataset did you use? You may want to try the whole thing again.

**Comment by 11mhg** [comment]
Hey! Thanks for the quick replies.

 I'm using the Coco dataset with just a small amount of labels chosen. When I warm it up, it starts at a decent recall and goes down to zero. I load the warmed up weights and then do training proper and it seems that the recall starts at zero and never goes up. I'm currently training again with some modified parameters, so I'll let you know how that goes, but if you happen to have insight on Coco, that would be great!

Thanks again!

**Comment by experiencor** [comment]
@11mhg The warmup training looks right to me, but the actual training is odd. You may try to train the detector for just one class to see how it goes.

**Comment by 11mhg** [comment]
So what I'm finding is that the recall converges quickly to zero during warmup, but when I do the training it very slowly increases recall. So for the one label "person", after 56 or so epochs I get a recall average of 0.0085..... What is a normal recall average I should get?

**Comment by experiencor** [comment]
@11mhg Current recall (an estimate of mAP) should be more than 0.3 for good detections in my experience. I assume that you have carefully checked the labels of the images.

**Comment by 11mhg** [comment]
Yes, I've verified the labels and done a multitude of tests on the parse annotations to make sure it is correct. I will take a look at using perhaps an older COCO dataset to see what happens.

**Comment by 11mhg** [comment]
okay, what I've found is that simply using the weights trained on COCO 2014 form darknet, I managed to fine tune on the COCO 2017 dataset with a recall of around 0.34!

**Comment by txdat** [comment]
@anvenkat09 can you tell me what you do, i have same problem with resnet

**Comment by tamersalama** [comment]
Trying to update the nan related issues - what worked for me is adding images and annotations added to "valid_image_folder" (I previously relied on having the training ones split 80/20 as per the readme - but I got nan for losses). I also changed the traing nb_epochs to 10 - and will likely need more (from 1)
